cassiopeia/solver/CCPhyloSolver.py

```diff
@@ -0,0 +1,223 @@
+
+        #check that ccphylo_path is valid
+        if not os.path.exists(self.ccphylo_path):
+            raise DistanceSolverError(
+                f"ccphylo_path {self.ccphylo_path} does not exist. To use fast "
+                "versions of Neighbor-Joining and UPGMA please install "
+                "CCPhylo (https://bitbucket.org/genomicepidemiology/ccphylo/src/master/) "
+                "set the ccphylo_path in the config.ini file then reinstall Cassiopeia."
+            )
+        
+        #check that ccphylo_path is executable
+        if not os.access(self.ccphylo_path, os.X_OK):
+            raise DistanceSolverError(
+                f"ccphylo_path {self.ccphylo_path} is not executable. To use fast "
+                "versions of Neighbor-Joining and UPGMA please install "
+                "CCPhylo (https://bitbucket.org/genomicepidemiology/ccphylo/src/master/) "
+                "set the ccphylo_path in the config.ini file then reinstall Cassiopeia."
+            )
+    
+    def _fast_solve(
+        self,
+        cassiopeia_tree: CassiopeiaTree,
+        layer: Optional[str] = None,
+        collapse_mutationless_edges: bool = False,
+        logfile: str = "stdout.log",
+    ) -> None:
+        """Solves a tree using fast distance-based algorithms implemented by 
+        CCPhylo. To call this method the CCPhlyo package must be installed 
+        and the ccphylo_path must be set in the config file. The method attribute
+        specifies which algorithm to use. The function will update the `tree`.
+
+        Args:
+            cassiopeia_tree: CassiopeiaTree object to be populated
+            layer: Layer storing the character matrix for solving. If None, the
+                default character matrix is used in the CassiopeiaTree.
+            collapse_mutationless_edges: Indicates if the final reconstructed
+                tree should collapse mutationless edges based on internal states
+                inferred by Camin-Sokal parsimony. In scoring accuracy, this
+                removes artifacts caused by arbitrarily resolving polytomies.
+            logfile: File location to log output. Not currently used.
+        """
+
+        self._setup_ccphylo()
+
+        dissimilarity_map = self.get_dissimilarity_map(cassiopeia_tree, layer)
+
+        with tempfile.TemporaryDirectory() as temp_dir:
+
+            # save dissimilarity map as phylip file
+            dis_path = os.path.join(temp_dir, "dist.phylip")
+            tree_path = os.path.join(temp_dir, "tree.nwk")
+            solver_utilities.save_dissimilarity_as_phylip(dissimilarity_map, dis_path)
+
+            # run ccphylo
+            command = f"{self.ccphylo_path} tree -i {dis_path} -o {tree_path} -m {self.method}"
+            process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            stdout, stderr = process.communicate()
+            T = ete3.Tree(tree_path, format=1)
```

Comment:
Is there a reason here why we can't read in the newick file with the Cassiopeia API? We have a few functions implemented under `tree.data.utilities` for reading in newick trees (and you can also pass the newick string to the `CassiopeiaTree` constructor)

Reply:
I want to use a few ete3 methods so I'm sticking with the ete3 reader, but I've switched to using `tree.data.utilities.ete3_to_networkx` to convert the ete3 tree to the nx tree